doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,192 @@
+Note WinHTTP transport adapter is only supported on Windows.
+
+Another example is if you want to create your own HTTP transport adapter for some other C++ HTTP stack. In this case, you can build your own HTTP transport adapter and test it by comparing it to the behavior of the libcurl HTTP transport adapter.
+
+## Using the HTTP transport adapter
+The HTTP transport adapter is set up during service SDK client initialization, for example, when creating an Azure Storage SDK client. The HTTP transport adapter can be specified at client initialization via the client options argument. See the following code snippet as an example:
+```cpp
+  /*
+  * Using the HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one based on the compiler options passed in to CMake.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+
+  // Use client options to set an HTTP transport adapter, which needs to be a shared_ptr.
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+### Re-use the HTTP transport adapter 
+Note that the HTTP transport adapter is a `shared_ptr`. This is because you can re-use the same HTTP transport adapter for multiple clients. There are two ways of doing this. The first one is by using the same HTTP transport adapter when creating another client. The second one is when you get a specific child client from an already created parent client directly, in which case the child client inherits the same configuration. See the next example:
+
+```cpp
+  /* 
+  *  Option 1. Init a new client with the same HTTP transport adapter.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, optionsA);
+  // The second client.
+  BlobClientOptions optionsB;
+  optionsB.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientB = BlobServiceClient(url, credential, optionsB);
+
+  /* 
+  *  Option 2. Create new client from a parent client.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClient = BlobServiceClient(url, credential, options);
+  // Create specific child client from the parent service client.
+  auto blobContainerClient = storageClient.GetBlobContainerClient(containerName)
+```
+
+Note that the second option may not supported for some Azure SDK clients if it is not applicable depending on the service behavior (for example, the Key Vault client).
```

Comment:
Just delete this sentence as I think it's obvious if you can't do it, you can't do it.